File: Ruby/columns.rb
#!/usr/bin/ruby

def print_column(result, result_counter)
  if result_counter % $COLUMNS != 0
    $stdout.print(" ")
  end
  result_counter = result_counter + 1
  $stdout.print($RESULT_FORMAT % result)
  if result_counter % $COLUMNS == 0
    $stdout.print("\n")
  end
  return result_counter
end


def finish_column(result_counter)
  remainder = result_counter % $COLUMNS
  if remainder != 0
    $stdout.print("\n")
  end
end

File: Python/columns.py
#!/usr/bin/python

import sys

def print_column(columns, result_format, result, result_counter):
    if result_counter % columns != 0:
        sys.stdout.write(" ")
    result_counter += 1
    sys.stdout.write(result_format % (result))
    if result_counter % columns == 0:
        sys.stdout.write("\n")
    return result_counter


def finish_column(columns, result_counter):
    if result_counter % columns != 0:
        sys.stdout.write("\n")

